e2e/tests/attendance-tests.spec.ts

```diff
@@ -131,6 +131,94 @@ test("Recurring activities list", async ({ page }) => {
-  await expect(page.getByRole("row")).toHaveCount(10);
+  await expect(page.getByRole("row")).toHaveCount(3);
   await argosScreenshot(page, "recurring-activities-list");
 });
+
+test("Edit participants of a recurring activity", async ({ page }) => {
+  const [user] = generateUsers();
+  const childToAdd = generateChild({
+    name: "AAAA", // FIXME: This ensure that the child is listed first
+  });
+  const childToKeep = generateChild();
+  const childToRemove = generateChild();
+  const allChildren = [
+    childToAdd,
+    childToKeep,
+    childToRemove,
+    ...times(6, () => generateChild()),
+  ];
+
+  const coachingClass = generateActivity({
+    title: "<COACHING CLASS>",
+    participants: [childToKeep, childToRemove],
+    assignedUser: user,
+  });
+
+  const otherActivities = times(3, () =>
+    generateActivity({
+      participants: faker.helpers.arrayElements(allChildren),
+      assignedUser: user,
+    }),
+  );
+
+  await loadApp(page, [
+    user,
+    ...allChildren,
+    ...otherActivities,
+    coachingClass,
+  ]);
+
+  // When I click on Attendance from the main menu
+  await page.getByRole("navigation").getByText("Attendance").click();
+
+  // And I click on "Manage Activities"
+  await page.getByRole("button", { name: "Manage Activities" }).click();
+
+  // And I Click on "<COACHING CLASS>"
+  await page.getByRole("cell", { name: "<COACHING CLASS>" }).click();
+
+  // And I click on the "Participants" tab
+  await page.getByRole("tab", { name: "Participants" }).click();
+
+  // And I click on "Edit"
+  await page.getByRole("button", { name: "Edit" }).click();
+
+  // And I click on the "Participants" field
+  await page.getByLabel("Participants").click();
+
+  // Then I see "Abhisyanta Sharma" selected.
+  await expect(
+    page
+      .getByRole("option", { name: childToKeep.name })
+      .getByRole("checkbox", { checked: true }),
+  ).toBeVisible();
+
+  // When I unselect "Abhisyanta Sharma."
+  await page.getByRole("option", { name: childToRemove.name }).click();
+
+  // And I select "Aasha Gill"
+  await page.getByRole("option", { name: childToAdd.name }).click();
+
+  // And I click "Save."
+  await page.getByRole("button", { name: "Save" }).click();
+
+  // Then the "Participants" field contains "Aasha Gill"
+  await expect(page.getByText(childToAdd.name)).toBeVisible();
+
+  // And the "Participants" field does not contain "Abhisyanta Sharma."
+  await expect(page.getByText(childToRemove.name)).not.toBeVisible();
+
+  // When I navigate to the Record Attendance screen
+  await page.getByRole("navigation").getByText("Attendance").click();
+  await page.getByRole("button", { name: "Record" }).click();
+
+  // And I click "Show more"
+  await page.getByRole("button", { name: "Show more" }).click();
+
+  // And I click "Coaching Class 3M"
+  await page.getByText("<COACHING CLASS>").click();
+
+  // Then I can record attendance for "Aasha Gill"
+  await expect(page.getByText(childToAdd.name)).toBeVisible();
```

Comment:
do we already have any screenshot of recording attendance flow? If not, take one here?